library/Icinga/Web/Navigation/Mobile/MobileNavigation.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/* Icinga Web 2 | (c) 2022 Icinga GmbH | GPLv2+ */
+
+namespace Icinga\Web\Navigation\Mobile;
+
+use Icinga\Authentication\Auth;
+use Icinga\Module\Director\Web\Table\Dependency\Html;
+use ipl\Html\Attributes;
+use ipl\Html\BaseHtmlElement;
+use ipl\Html\HtmlElement;
+use ipl\Html\HtmlString;
+use ipl\Html\Text;
+use ipl\Web\Widget\Icon;
+
+class MobileNavigation extends BaseHtmlElement
+{
+    const MAX_NUM_OF_ITEMS = 5;
+
+    protected $tag = 'nav';
+
+    protected $defaultAttributes = [
+        'id' => 'mobile-menu',
+        'class' => 'container',
+        'data-base-target' => '_main'
+    ];
+
+    protected $items;
+
+    public function __construct()
+    {
+        $this->items = [
+            'dashboard' => [
+                'icon' => 'th',
+                'title' => t('Dashboard')
+            ],
+            'problems' => [
+                'icon' => 'exclamation-circle',
+                'title' => t('Problems')
+            ],
+            'icingadb' => [
+                'icon' => 'database',
+                'title' => t('IcingaDB')
+            ],
+            'overview' => [
+                'icon' => 'binoculars',
+                'title' => t('Overview')
+            ],
+            'history' => [
+                'icon' => 'history',
+                'title' => t('History')
+            ],
+            'test' => [
+                'icon' => 'vials',
+                'title' => t('Test')
+            ],
+            'test2' => [
+                'icon' => 'vial',
+                'title' => t('Test 2')
+            ],
+        ];
+    }
+
+    protected function createMoreItem()
+    {
+        $moreMenu = new HtmlElement('li');
+
+        $this->assembleMoreItem($moreMenu);
+
+        return $moreMenu;
+    }
+
+    protected function assembleMoreItem(BaseHtmlElement $moreMenu)
+    {
+        $moreMenu->add(
+            HtmlElement::create(
+                'button',
+                Attributes::create(['id' => 'toggle-more']),
+                [
+                    new Icon('ellipsis-v'),
+                    Text::create('More')
+                ]
+            )
+        );
+
+        $moreMenu->add($this->createMoreFlyout());
+    }
+
+    protected function createMoreFlyout()
+    {
+        $moreFlyout = new HtmlElement('div', Attributes::create(['class' => 'flyout']));
+
+        $this->assembleMoreFlyout($moreFlyout);
+
+        return $moreFlyout;
+    }
+
+    protected function assembleMoreFlyout($moreFlyout)
+    {
+        $flyoutContent = HtmlElement::create('div', ['class' => 'flyout-content']);
+        $ul = HtmlElement::create('ul', ['class' => 'nav level-2']);
+
+        foreach ($this->items as $name => $item) {
```

Comment:
You are adding all items again in moreFlyover. Only items that are not added to the main navigation should be added here.